File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        unordered_set<char> mySet;
        int i = 0, ans = 0, n = s.size();
        for(int j = 0; j < n; j++) {
            while(mySet.find(s[j]) != mySet.end()) {
                mySet.erase(s[i]);
                i++;
            }
            mySet.insert(s[j]);
            ans = max(ans, (j-i)+1);
        }
        return ans;
    }
};

File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        freq = {}
        i = 0
        ans = 0
        for j in range(len(s)):
            if s[j] not in freq:
                freq[s[j]] = 1
            else :
                freq[s[j]] += 1
                
            while (freq[s[j]] > 1): 
                freq[s[i]] -= 1
                i += 1
            ans = max(ans, (j-i)+1)
        return ans
